frontend/packages/dev-console/src/utils/pipeline-plugin-utils.ts

```diff
@@ -0,0 +1,71 @@
+import * as _ from 'lodash';
+import { referenceForModel, K8sResourceKind } from '@console/internal/module/k8s';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { PipelineRunModel, PipelineModel } from '../models';
+import { Pipeline, PipelineRun } from './pipeline-augment';
+
+// label to get the pipelines
+export const PIPELINE_REF = 'app.openshift.io/pipeline-ref';
+
+export const tknPipelineAndPipelineRunsResources = (namespace: string): FirehoseResource[] => {
+  const resources = [
+    {
+      isList: true,
+      kind: referenceForModel(PipelineRunModel),
+      namespace,
+      prop: 'pipelineRuns',
+      optional: true,
+    },
+    {
+      isList: true,
+      kind: referenceForModel(PipelineModel),
+      namespace,
+      prop: 'pipelines',
+      optional: true,
+    },
+  ];
+  return resources;
+};
+
+type PipelineItem = {
+  pipeline: K8sResourceKind;
+  pipelineRuns: K8sResourceKind[];
+};
+
+const byCreationTime = (left, right) => {
+  const leftCreationTime = new Date(_.get(left, ['metadata', 'creationTimestamp'], Date.now()));
+  const rightCreationTime = new Date(_.get(right, ['metadata', 'creationTimestamp'], Date.now()));
+  return rightCreationTime.getTime() - leftCreationTime.getTime();
+};
+
+const getPipelineRunsForPipeline = (pipeline: Pipeline, props): PipelineRun[] => {
+  if (!props || !props.pipelineRuns) return null;
+  const pipelineRunsData = props.pipelineRuns.data;
+  const PIPELINE_RUN_LABEL = 'tekton.dev/pipeline';
+  const pipelineName = pipeline.metadata.name;
+  return pipelineRunsData
+    .filter((pr: PipelineRun) => {
+      return (
+        pipelineName ===
+        (_.get(pr, ['spec', 'pipelineRef', 'name'], null) ||
+          _.get(pr, ['metadata', 'labels', PIPELINE_RUN_LABEL], null))
+      );
+    })
+    .sort(byCreationTime);
```

Comment:
Any reason for the `sort`?

Reply:
Yes, to have the latest run at start of the list because we want to show 3 latest runs on the side panel.

Reply:
I wonder if this is not more suited for your usage of the code, rather than at the top level.

Reply:
Consider the view point of possible usages:
 - Would we want PipelineRuns for a table view - where the user can sort?
 - Would we want PipelineRuns for a filterable view?

It's not always beneficial to apply mutations like sort on a low-level set of data.

Reply:
A default sort can't hurt though.

Reply:
Perhaps. If you're not bothered by it, I'll let it go.